plugins/cookie_manager/lib/src/cookie_mgr.dart

```diff
@@ -28,7 +28,25 @@ class CookieManager extends Interceptor {
   final CookieJar cookieJar;
 
   /// Merge cookies into a Cookie string.
+  /// Cookies with longer paths are listed before cookies with shorter paths.
   static String getCookies(List<Cookie> cookies) {
+    // Remove previous cookie with same name.
+    cookies = cookies.fold<List<Cookie>>([], (prev, current) {
+      // Check if the current cookie's name is already in the list.
+      if (prev.any((c) => c.name == current.name)) {
+        // Skip this cookie.
+        return prev;
```

Comment:
Deeply reviewing these codes, I think it'll be better if we use `[...previousCookies, ...cookies]` and return the new cookie here. Even though they are the same theoretically at present, it still causes confusion when reading these codes.

Reply:
Sorry I am a bit confuse here, is it the **.fold** that causing the confusion?
If that is the case, we may use a Set variable to avoid the cookie with duplicate name.
~~~
 final uniqueNames = <String>{};
 // Filter out duplicate cookie name.
 cookies = cookies.where((cookie) => uniqueNames.add(cookie.name)).toList();
~~~
Is this has better readability?

Reply:
https://github.com/cfug/dio/blob/6b1f9edcfb575a6f4fdfc588cd5170e62bb04439/plugins/cookie_manager/lib/src/cookie_mgr.dart#L48-L54

Here we placed new cookies at the front, can we consider placing previous cookies first?1
```dart
 final newCookies = getCookies([
   ...?previousCookies
       ?.split(';')
       .where((e) => e.isNotEmpty)
       .map((c) => Cookie.fromSetCookieValue(c)),
   ...cookies,
 ]);
```

Reply:
This was working before we added the sorting logic, I have no idea why since the order should be remained the same with all are having the same path. Let me further dig out why it is not working and update here later.

Reply:
I brought this up because this is the change that happened in your first commit.

Reply:
Indeed that was my first commit, and I found why it is not working with the sorting logic.
The previous cookies are constructed with path = null, and the sorting logic will place it to the last in order. 
To fix:
1. Fix the null path issue, but the previous cookies are retrieved from the header which has no path info.
2. Sort the null path in front.
3. Remove duplicate name.

What do you think?

Reply:
Leaving the path as not nullable can be good in this case but good in compatibilities. The thing is `getCookies` is a public API, which can be used with `[...previous, ...new]` by users which makes it valid.

Reply:
Meanwhile, I will fix with option #2, which sorts the null path in front. Is it ok for you?

Reply:
> Meanwhile, I will fix with option #2, which sorts the null path in front. Is it ok for you?

Yeah sure. Thanks!

Reply:
> Leaving the path as not nullable can be good in this case but good in compatibilities.

Sorry about the typos, here I mean we should correct the `getCookies` rather than change fields like paths.